sklearn/manifold/tests/test_t_sne.py

```diff
@@ -1277,10 +1279,24 @@ def test_tsne_deprecation_square_distances():
         n_components=2,
         init="pca",
         learning_rate="auto",
-        perplexity=30.0,
+        perplexity=25.0,
         angle=0,
         n_jobs=1,
         random_state=0,
     )
     X_trans_2 = tsne.fit_transform(X)
     assert_allclose(X_trans_1, X_trans_2)
+
+
+@pytest.mark.filterwarnings("ignore:The default learning rate in TSNE")
+@pytest.mark.filterwarnings("ignore:The default initialization in TSNE")
+def test_tsne_perplexity_validation():
+    """Make sure that perplexity > n_samples results in a ValueError"""
+
+    random_state = check_random_state(0)
+    X = random_state.randn(20, 2)
+    TSNE(perplexity=5).fit_transform(X)
+    msg = "perplexity must be less than n_samples"
+    with pytest.raises(ValueError, match=msg):
+        TSNE(perplexity=20).fit_transform(X)
+        TSNE(perplexity=30).fit_transform(X)
```

Comment:
This check never checks 30, since the call with `perplexity=20` will always fail. I think we need to `pytest.mark.parametrize` to test both cases.